### DISCUSSION TITLE
Can we limit /events/_search to Organizations?

### DISCUSSION THREAD

**Discussion by nikzen** [zitadel/zitadel#5881] [General]
Heyhey,

we are using the POST /events/_search to listen to all events and trigger workflows based on the event. To use the API a service user needs IAM_OWNER_VIEW rights and is then able to get all events from the instance. 

However, we want to limit the output of events to organizations so that we do not leak metadata of other organizations to an organization.

We are a little bit unsure how we can solve this issue. Do you have an idea?

**Comment by hifabienne** [comment]
Hei @nikzen 
At the moment this endpoint is only included in the admin api. which means it is overall organizations.
You can send the resouceowner in the request to only get data from a specific organization. resource owner = organization, so you can send the organization id there.

**Reply by nikzen** [reply]
Yes, but it does not really solve the issue that theoretically we can leak data. Another idea would be to implement the events endpoint on on the management api. However we are unsure about the effort.

**Reply by hifabienne** [reply]
We are currently building our new ressources API, where we do not differ between the different contexts of managament, auth, admin, etc.
Our goal is to have one API, where you can call ressources and depending on the requesting user and the permission he has, we can send back more or less.
The api will start with the ressources that are needed for registration and login. After that we will start integrating more and more. 
At one point also the events request will be included there.

So for the moment what you try to do is not directly possible, but it will in the future.

You can find some more information about the api here: https://github.com/zitadel/zitadel/discussions/5875
